### PR TITLE
[WIP] Spectral normalization layer

### PR DESCRIPTION
**Pull request by hartikainen**
Implements SpectralNormalization layer [1].

This is still work in progress, but would be great to get some feedback on this. Things that still need to be done:
- [ ] Make sure that the data shapes and axes are handled correctly
- [ ] Make sure that the `u` and `layer.kernel` assignments are handled correctly
- [ ] Test the wrapper

[1] Miyato, T., Kataoka, T., Koyama, M., & Yoshida, Y. (2018). Spectral normalization for generative adversarial networks. arXiv preprint [arXiv:1802.05957](https://arxiv.org/pdf/1802.05957.pdf).

### CODE REVIEW

**Comment by Squadrick** [comment]
Hey @hartikainen thanks for the contribution. The code looks great so far. Just a reminder, you'll need to update `BUILD`, `__init__.py` and `README.md` as well.

**Comment by 183amir** [comment]
How about the spectral normalization implementation that is in tf-gan? https://github.com/tensorflow/gan/blob/master/tensorflow_gan/python/features/spectral_normalization.py

**Comment by charlielito** [comment]
> How about the spectral normalization implementation that is in tf-gan? https://github.com/tensorflow/gan/blob/master/tensorflow_gan/python/features/spectral_normalization.py
I think that implementation is straight forward compatible with tf.keras layers and also tf2.0
What happened with this PR?

**Comment by gabrieldemarmiesse** [comment]
Closing as this pull request had no activity for a while.
